tests/integration/element__click_test.py

```diff
@@ -0,0 +1,35 @@
+from tests.integration.helpers.givenpage import GivenPage
+
+
+def test_click_waits_for_covering_preload_element_to_disappear(session_browser):
+    page = GivenPage(session_browser.driver)
+    page.opened_with_body(
+        ''' <style>
+             canvas {
+              background-color: rgba(0, 0, 0, 0.336);
+              position: fixed;
+              height: 100%;
+              width: 100%;
+              top: 0;
+              left: 0;
+              z-index: 1000;
+             }
+            </style>
+
+            <a href="#second">go to Heading 2</a>
+            <h2 id="second">Heading 2</h2>       
+
+            <p><canvas></canvas></p>'''
+    )
+    page.load_body_with_timeout(
+        '''
+        <a href="#second">go to Heading 2</a>
+        <h2 id="second">Heading 2</h2>       
```

Comment:
this part duplicates the corresponding lines from previous call, inside triple-quotes string... 
Also, taking into account that the part of our logic is "waiting for overlay to be hidden" - it would be reasonable to emphasize this in code – not rewrite html body, but change the display property to `none`... 

Something like this:

```
    page.execute_script_with_timeout(
        '''
        document.getElementById('overlay').style.display = 'none';  
        ''',
        250,
    )
```

But there is one problem, ther is no `execute_script_with_timeout` inside GivenPage
<img width="713" alt="image" src="https://user-images.githubusercontent.com/1961118/114300166-8d9d5580-9ac7-11eb-8c49-63bec3295a19.png">

Yet, we have it in
<img width="918" alt="image" src="https://user-images.githubusercontent.com/1961118/114300170-968e2700-9ac7-11eb-9967-593a5d1b4303.png">

All this code is pretty complicated, the time now is not the one to refactor it. But you can simply create an alias for this method inside GivenPage, same way as other methods are implemented;)